backend/src/model/entity/plantings.rs

```diff
@@ -1,10 +1,23 @@
 //! All entities associated with [`Planting`].
 
+use chrono::NaiveDate;
 use diesel::{AsChangeset, Identifiable, Insertable, Queryable};
 use uuid::Uuid;
 
 use crate::schema::plantings;
 
+/// Arguments for the database layer find plantings function.
+pub struct FindPlantingsParameters {
+    /// The id of the plant to find plantings for.
+    pub plant_id: Option<i32>,
+    /// The id of the layer to find plantings for.
+    pub layer_id: Option<i32>,
+    /// First date in the time frame plantings are searched for.
+    pub from: NaiveDate,
+    /// Last date in the time frame plantings are searched for.
+    pub to: NaiveDate,
+}
+
```

Comment:
I personally feel like you should just pass them in directly. This is not really and entity so it doesn't fit here. If you want the struct you should probably move it to where it is actually used.

Reply:
@kitzbergerg
I added them here so that the entity layer does not depend on the service layer but the other way round.
Would moving it to the impl be better?
I do not want to pass it in, because conceptually the service should handle the business logic, not the entity.

Reply:
Another reason why I introduced a separate struct is the result error type of the entity functions.
This one produces a ServiceError.
```rs
let from = search_parameters
        .relative_to_date
        .checked_sub_days(Days::new(TIME_LINE_LOADING_OFFSET_DAYS))
        .ok_or_else(|| {
            ServiceError::new(
                StatusCode::BAD_REQUEST,
                "Could not add days to relative_to_date".into(),
            )
        })?;
```
The entities return a DieselError.